### ISSUE TITLE
No Interfaces can only extend interfaces without "= function () {}"

### ISSUE THREAD

**Issue by zavr-1** [google/closure-compiler#3522]
I want an interface to extend a constructor. It's always been working fine like so
```js
/**
 * The application interface.
 * @extends {events.EventEmitter}
 * @interface
 */
_goa.Application
```
This did behave indeed like an interface, I could annotate JS code like `@implements {_goa.Application}`. 
I then added the assignment `= function() {}`
```js
/**
 * The application interface.
 * @extends {events.EventEmitter}
 * @interface
 */
_goa.Application = function() {}
```
This started giving me the warning 
```
types/externs/application.js:13: WARNING - [JSC_CONFLICTING_EXTENDED_TYPE]
_goa.Application cannot extend this type; interfaces can only extend interfaces
```

I think you should allow to extend constructors because the two are closely related, and tsicle will generate constructors and not interfaces. The difference is pretty subtle anyway, there's not interface concept in JS so I intuitively know what the difference is but it's hard to explain, it can be only explained in terms what you can and cannot do with both, like write `@implements` for interfaces.

https://github.com/google/closure-compiler/issues/937 this was closed 

So this issue, is that there's no `interfaces can only extend interfaces` check on interfaces without `= function() {}` in externs, but please don't add it because say Node.JS API has EventEmitter, which is a constructor (maybe I should actually rename it to interface, like here I struggle to understand what the difference is, it's an implementation and and interface at the same time, so why should't I be able to create an interface which extends it... in any way, you get the idea.

**Comment by zavr-1** [comment]
plus interfaces can extend records without a warning which is inconsistent with the statement "interfaces can only extend interfaces"

**Comment by lauraharker** [comment]
> So this issue, is that there's no interfaces can only extend interfaces check on interfaces without = function() {} in externs

Yes, that's what's happening. The "interfaces can only extend interfaces" check assumes that all interface declarations have an associated function node. This was definitely not intentional on our part, though.

The only call to [`checkInterface`](https://github.com/google/closure-compiler/blob/88f4418a5d8ffa3958aea063601f90f50c8ed771/src/com/google/javascript/jscomp/TypeCheck.java#L2470) is within the function visiting `FUNCTION` nodes:
https://github.com/google/closure-compiler/blob/88f4418a5d8ffa3958aea063601f90f50c8ed771/src/com/google/javascript/jscomp/TypeCheck.java#L2364

The main difference between constructors & interfaces (at least to Closure Compiler) is that constructors can be instantiated via `new` and exist at runtime, and interfaces cannot be instantiated and do not exist at runtime. If `EventEmitter` is new-able then it should definitely be a constructor.

If you need to have a non-instantiable subtype of `EventEmitter`, how about an @abstract class?  Or create a separate interface that both `EventEmitter` and your interface implement?
